src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.cs

```diff
@@ -52,14 +54,16 @@ public void Setup()
         }
 
         [Benchmark]
-        public async Task HandshakeAsync()
+        [Arguments(true)]
+        [Arguments(false)]
+        public async Task HandshakeAsync(bool useRsaCertificate)
```

Comment:
@wfurt do you think that we might want to use more certificates in the future? I just wonder if instead of using a boolean, we should have sth like:

```cs
[Benchmark]
public Task HandshakeAsync() => HandshakeAsync(_cert);

[Benchmark]
public Task HandshakeECDSACertAsync() => HandshakeAsync(_ecdsaCert);

private async Task HandshakeAsync(X509Certificate certificate)
{
    using (var sslClient = new SslStream(_client, leaveInnerStreamOpen: true, delegate { return true; }))
    using (var sslServer = new SslStream(_server, leaveInnerStreamOpen: true, delegate { return true; }))
    {
        await Task.WhenAll(
            sslClient.AuthenticateAsClientAsync("localhost", null, SslProtocols.None, checkCertificateRevocation: false),
            sslServer.AuthenticateAsServerAsync(certificate, clientCertificateRequired: false, SslProtocols.None, checkCertificateRevocation: false));

        // Workaround for corefx#37765
        await sslServer.WriteAsync(_serverBuffer, default);
        await sslClient.ReadAsync(_clientBuffer, default);
    }
}
```

which would make it easier to add new certificates in the future

Reply:
I'm not sure we will. But I do like the idea. I think it will also look better on the final report.

Reply:
updated.

|                         Method |         Mean |        Error |       StdDev |       Median |          Min |          Max |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------------------- |-------------:|-------------:|-------------:|-------------:|-------------:|-------------:|-------:|------:|------:|----------:|
|                 HandshakeAsync | 54,292.01 us | 1,335.331 us | 1,537.769 us | 54,604.22 us | 50,362.14 us | 56,005.38 us |      - |     - |     - |   20096 B |
|        HandshakeECDSACertAsync | 45,455.02 us | 1,006.735 us | 1,159.358 us | 45,322.94 us | 44,115.81 us | 47,581.19 us |      - |     - |     - |   11848 B |
|                 WriteReadAsync |     10.52 us |     0.202 us |     0.179 us |     10.50 us |     10.31 us |     11.01 us |      - |     - |     - |         - |
|                 ReadWriteAsync |     19.17 us |     0.413 us |     0.459 us |     19.11 us |     18.50 us |     20.23 us | 0.0800 |     - |     - |     344 B |
|            ConcurrentReadWrite |     13.16 us |     0.248 us |     0.275 us |     13.16 us |     12.54 us |     13.64 us |      - |     - |     - |       2 B |
| ConcurrentReadWriteLargeBuffer |     18.58 us |     0.904 us |     1.041 us |     18.70 us |     16.97 us |     20.52 us |      - |     - |     - |         - |

Reply:
Seems like `HandshakeAsync` should be `HandshakeRSACertAsync`, then.

Reply:
We use an HTTP certificate which happens to be RSA. Would it make sense to add dedicated certs for this test? 
Also from perf perspective, the difference is large. But there may be security implications. Would it make sense do have something where the security level is somewhat similar? 
I think it may be useful if we want to use the data for any guidance. 
The performance impact of a particular algorithm is much larger than I expected across platform. I'll probably add more tests later so we have better visibility.

Reply:
There's no such thing as an HTTP certificate :smile:.  I'm just saying that if the difference between `HandshakeAsync` and `HandshakeECDSACertAsync` is the cert it uses, then it's more obvious what the "default" one is doing if you say RSACert.

Though, looking at the implementation, I guess the resource it's loading doesn't explicitly say RSA, so maybe this is fine.

I'd probably use hard-coded keys and generate the cert/certs on the fly to avoid an expiration problem; at which point one is definitely RSA and one is definitely ECDSA.  But maybe that's just me :smile:.